pkg/promclient/promclient.go

```diff
@@ -335,6 +338,32 @@ func PromqlQueryInstant(ctx context.Context, logger log.Logger, base *url.URL, q
 	return vec, nil
 }
 
+// GetPromVersion will return the version of Prometheus by querying /version Prometheus endpoint.
+func GetPromVersion(logger log.Logger, base *url.URL) (*version.Version, error) {
```

Comment:
Golang does not like getters, so this should be fine.

```suggestion
func PromVersion(logger log.Logger, base *url.URL) (*version.Version, error) {
```

Reply:
Done.